### PR TITLE
Add Spring MVC sample to README

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
### Issue:

https://github.com/okta/okta-spring-boot/issues/178

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +150,31 @@ public class SecurityConfiguration {
 
 [Full Stack Reactive with Spring WebFlux, WebSockets, and React](https://developer.okta.com/blog/2018/09/25/spring-webflux-websockets-react) uses both SSO and a resource server. Its current code uses Spring Security's OIDC support. [Changing it to use the Okta Spring Starter](https://github.com/oktadeveloper/okta-spring-webflux-react-example/pull/11) reduces the lines of code quite a bit.
 
+### Spring MVC
+
+To configure a resource server when using Spring MVC, you need to extend the `WebSecurityConfigurerAdapter` bean and override the `configure` method.
+
+```java
+@Configuration
+public class OktaOAuth2WebSecurityConfigurerAdapter extends WebSecurityConfigurerAdapter {
```
Comment (bdemers) [inline README.md:159]: This is a Resource Server example, I would expect something for a more traditional MVC application in this example (similar to the example above)

**Review by bdemers — CHANGES_REQUESTED** [review]
see comment above

File: README.md (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply README.md]: updated.

**Comment by arvindkrishnakumar-okta** [comment]
> see comment above

fixed.

**Review by bdemers — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -150,6 +150,81 @@ public class SecurityConfiguration {
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@EnableWebSecurity
+public class OAuth2ResourceServerSecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http.authorizeRequests()
+            .antMatchers(HttpMethod.GET, "/message/**").hasAuthority("SCOPE_message:read")
```
[the remark this answers is not in the document]
Comment (arvindkrishnakumar-okta) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -150,6 +150,81 @@ public class SecurityConfiguration {
 
 [Full Stack Reactive with Spring WebFlux, WebSockets, and React](https://developer.okta.com/blog/2018/09/25/spring-webflux-websockets-react) uses both SSO and a resource server. Its current code uses Spring Security's OIDC support. [Changing it to use the Okta Spring Starter](https://github.com/oktadeveloper/okta-spring-webflux-react-example/pull/11) reduces the lines of code quite a bit.
 
+### Spring MVC
+
+1. Annotate your application main class with `@SpringBootApplication`.
```
[the remark this answers is not in the document]
Comment (arvindkrishnakumar-okta) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -150,6 +150,81 @@ public class SecurityConfiguration {
 
 [Full Stack Reactive with Spring WebFlux, WebSockets, and React](https://developer.okta.com/blog/2018/09/25/spring-webflux-websockets-react) uses both SSO and a resource server. Its current code uses Spring Security's OIDC support. [Changing it to use the Okta Spring Starter](https://github.com/oktadeveloper/okta-spring-webflux-react-example/pull/11) reduces the lines of code quite a bit.
 
+### Spring MVC
```
[the remark this answers is not in the document]
Comment (arvindkrishnakumar-okta) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -93,6 +93,85 @@ Okta's Spring Security integration will [parse the JWT access token](https://dev
...
+
+	@PostMapping("/message")
+	@PreAuthorize("hasAuthority('SCOPE_message:write')")
+	public String createMessage(@RequestBody String message) {
+		return String.format("Message was created. Content: %s", message);
+	}
+}
+```
+
+Note: `message:read` and `message:write` used above in `@PreAuthorize` are OAuth scopes. If you are looking
```
Comment (bdemers) [inline README.md:140]:
```suggestion
**NOTE**: `message:read` and `message:write` used above in `@PreAuthorize` are OAuth scopes. If you are looking
```

File: README.md [hunk 6]
```diff
@@ -93,6 +93,85 @@ Okta's Spring Security integration will [parse the JWT access token](https://dev
...
+to add custom scopes, refer to the documentation [here](https://developer.okta.com/docs/guides/customize-authz-server/create-scopes/).
```
Comment (bdemers) [inline README.md:141]:
```suggestion
to add custom scopes, refer to the [documentation](https://developer.okta.com/docs/guides/customize-authz-server/create-scopes/).
```

SEO is better when the link text isn't here (or so i'm told)

File: README.md [hunk 7]
```diff
@@ -93,6 +93,85 @@ Okta's Spring Security integration will [parse the JWT access token](https://dev
...
+ 
+3. Configure your Resource Server either for JWT or Opaque Token validation by extending the `WebSecurityConfigurerAdapter` class 
+and overriding the `configure` method.
```
Comment (bdemers) [inline README.md:144]: Mention what happens by default

**Review by bdemers — APPROVED** [review]
Looks good, one minor comment, just add a note in the RS section about what happens by default

File: README.md (hunk 7 above)
Comment (arvindkrishnakumar-okta) [reply README.md]: added.

**Comment by arvindkrishnakumar-okta** [comment]
> Looks good, one minor comment, just add a note in the RS section about what happens by default

addressed.

**Pull request merged**